libs/fetcher.js

```diff
@@ -23,7 +23,7 @@ function parseValue(value) {
         // the MAX_SAFE_INTEGER in javascript is 9007199254740991 which has a length of 16.
         // Using a length > 15 check is still safe and faster than trying to use another
         // library that supports big integers.
-        if (typeof ret === 'number' && value.length > 15) {
+        if (typeof ret === 'number' && String(value) !== String(ret)) {
```

Comment:
can we just use `Number.isSafeInteger()`, which will also cover `Number.Infinity` case reported in #151? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger